mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -999,18 +998,17 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
     }
 
     private fun showMissingPermissionsWarningIfNeeded() {
-        val tasksWithPermissions = mutableMapOf(PrefKeys.SEND_PHONE_STATE to Manifest.permission.READ_PHONE_STATE)
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            tasksWithPermissions[PrefKeys.SEND_WIFI_SSID] = Manifest.permission.ACCESS_FINE_LOCATION
-        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            tasksWithPermissions[PrefKeys.SEND_WIFI_SSID] = Manifest.permission.ACCESS_COARSE_LOCATION
-        }
-
-        val missingPermissions = tasksWithPermissions
+        val missingPermissions = BackgroundTasksManager.KNOWN_KEYS
             .filter { entry ->
-                prefs.getStringOrNull(entry.key)?.toItemUpdatePrefValue()?.first == true && !hasPermission(entry.value)
+                val requiredPermissions = BackgroundTasksManager.getRequiredPermissionsForTask(entry)
+                prefs.getStringOrNull(entry)?.toItemUpdatePrefValue()?.first == true &&
+                    requiredPermissions != null && !hasPermissions(requiredPermissions)
             }
-            .map { entry -> entry.value }
+            .mapNotNull { entry -> BackgroundTasksManager.getRequiredPermissionsForTask(entry) }
```

Comment:
Can also amend a `?.toList()` for each array instead of the `toTypedArray()` below.

Reply:
Sure, makes it better readable.